base/tuple.jl

```diff
@@ -105,7 +105,8 @@ julia> ntuple(i -> 2*i, 4)
 ```
 """
 function ntuple(f::F, n::Integer) where F
-    t = n <= 0  ? () :
+    (n >= 0) || throw(DomainError())
```

Comment:
`ArgumentError`? A quick search through base suggests `DomainError` is mostly used when the value is out of domain for mathematical functions (e.g. `sqrt`, `log` etc)